### ISSUE TITLE
Programme image 

### ISSUE THREAD

**Issue by fulanumoto** [marcelGoerentz/Threadfin#34]
I'm using Threadfin with Jellyfin and have integrated the [Japanterebi](https://github.com/Animenosekai/japanterebi-xmltv) XMLTV file. Everything works well, but it would be perfect if it could display the progmme images instead of just the channel logos.

**Comment by marcelGoerentz** [comment]
Hey, are there links within the xml or within the m3u file?
If so, I think it will be possible to add an option where you can select between the channel logo or a program image. It would be helpful if you could share a file. Thanks in advance.

**Comment by fulanumoto** [comment]
@marcelGoerentz yes there are in the xml file `https://animenosekai.github.io/japanterebi-xmltv/guide.xml`

**Comment by marcelGoerentz** [comment]
Sorry for my late response, but do you want to show the program image in Jellyfin?

**Comment by fulanumoto** [comment]
> Sorry for my late response, but do you want to show the program image in Jellyfin?

No worries. Yes in Jellyfin Live TV.

**Comment by marcelGoerentz** [comment]
This is something that Jellyfin should do as it has all the information needed. The reason why I say this is that the program image should change as soon as the program changes. In my opinion Threadfin is only for proxying and buffering the streams. This would be a huge change within the database that I’m right now not capable to do. For this I have to rewrite the whole Database to provide the routines and data structure.

Sorry but right now I can’t implement this feature.

**Comment by fulanumoto** [comment]
You're absolutely right. I tested with another guide, and it turns out that Jellyfin only recognizes the `<icon>` tag, not the `<image>` tag. Thank you for your insight!

**Comment by fulanumoto** [comment]
Hi @marcelGoerentz 

Apologies for reopening this, but I’ve run into an issue where the program images are not being included in the `threadfin.xml` file generated by animenosekai.

It seems like the XML file itself is generated correctly, but the program image is missing. I’ve checked that the structure follows what’s expected, but the image URL /tag populated in the output is only for the channels logos.

Here’s what I’ve verified so far:

- The XML structure seems correct according to the expected format for Japanterebi.xmltv.
- Program image URLs are present in the source data, but they aren't showing up in the generated file.

Could this be related to a bug or something that might need additional configuration? Let me know if there are any troubleshooting steps I can follow or if more details are needed.

Thanks for your help!

**Comment by marcelGoerentz** [comment]
Ah I understand, I will take a look at this, maybe I need to add that this info will be saved also.

**Comment by marcelGoerentz** [comment]
Please test the latest beta version.
#88 should add this feature

**Comment by fulanumoto** [comment]
Thanks for the update Marcel! After testing the beta version, I can confirm that the image program is now successfully included in the generated Threadfin file. However, I noticed that Jellyfin and Tivimate only recognize the `<icon>` tag instead of  `<image>` . Would it be possible to change the tag from `<image>` to `<icon>` so that the image displays correctly as an icon in these platforms, instead of the channel logo?

Thanks again for your work on this!

**Comment by marcelGoerentz** [comment]
Ah I see...
This seems to be an issue with the XMLTV file you provided, since it delivers the image tag for some programs and I was just copying it.
I can transfer the image tag into an icon tag when this occurs.

**Comment by fulanumoto** [comment]
Thanks for the quick response, Marcel!

You're right, the XMLTV file I provided uses the `<image>` tag for the program images, not the `<icon>` tag. I appreciate you handling the transfer to the icon tag when that’s the case.

I’m happy to test the beta again once this update is made to verify everything works as expected. Thanks again for your help!

**Comment by marcelGoerentz** [comment]
Should be fixed in the newest beta: #89

**Comment by fulanumoto** [comment]
Thanks for the fix, Marcel!

I tested the update, and it works as expected, but I had to change the Stream Buffer setting to **No Buffer** instead of using ffmpeg to get the program images to show properly.

I also noticed that there are duplicates of the programs—one with the correct show name with its poster and another labeled as "dummy program."

![Image](https://github.com/user-attachments/assets/c016ad41-d469-48ab-b9fb-d0f3475d6994)

**Comment by marcelGoerentz** [comment]
The dummy programs should only show up when there is a time gap between 2 programs.
I will check if I can find a better solution.

**Comment by marcelGoerentz** [comment]
Fixed in #90

**Comment by fulanumoto** [comment]
Thank you so much for your help and for looking into this! I really appreciate your insights and the time you took to discuss the issue. Your suggestions have been super helpful. 
I'll go ahead and close the issue now. Thanks again!
